packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -605,9 +596,11 @@ class LocalizationsGenerator {
   });
 
   final FileSystem _fs;
-  Iterable<Message> _allMessages;
-  AppResourceBundleCollection _allBundles;
-  LocaleInfo _templateArbLocale;
+  Iterable<Message> _allMessages = <Message>[];
+  late final AppResourceBundleCollection _allBundles = AppResourceBundleCollection(inputDirectory);
+
+  late final AppResourceBundle _templateBundle = AppResourceBundle(templateArbFile);
+  late final LocaleInfo _templateArbLocale = _templateBundle.locale;
```

Comment:
These were set in `loadResources`.  Move them to `late final`s so they can be nonnull and lazily loaded.